xgbserver/rockcraft.yaml

```diff
@@ -0,0 +1,78 @@
+# Based on https://github.com/kserve/kserve/blob/master/python/xgb.Dockerfile
+# 
+# See ../CONTRIBUTING.md for more details about the patterns used in this rock. 
+# This rock is implemented with some atypical patterns due to the native of the upstream
+# Dockerfile.
+name: xgbserver
+summary: xgboost server for Kserve deployments
+description: "Kserve xgboost server"
+version: "0.11.2"
+license: Apache-2.0
+base: ubuntu@22.04
+platforms:
+    amd64:
+run-user: _daemon_
+services:
+  xgbserver:
+    override: replace
+    summary: "xgbserver server service"
+    startup: enabled
+    command: "python -m xgbserver [ ]"
+entrypoint-service: xgbserver
+
+parts:
+  security-team-requirement:
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && \
+      dpkg-query --root=${CRAFT_PROJECT_DIR}/../bundles/ubuntu-22.04/rootfs/ -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) \
+      > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  python:
+    plugin: nil
+    source: https://github.com/kserve/kserve.git
+    source-subdir: python
+    source-tag: v0.11.2
+    build-packages:
+      - libgomp1
+    overlay-packages:
+    - python3.10  
+    # Including python3-pip here means pip also gets primed for the final rock
+    - python3-pip
+    override-build: |
+      # Populate the build system's python environment with all packages needed for 
+      # the server in the final rock
+      
+      # Setup poetry
+      pip install poetry==1.4.0
+      poetry config virtualenvs.create false 
+
+      # Install the kserve package, this specific server package, and their dependencies.
+      mkdir -p ./python_env_builddir
+      cp -rf $CRAFT_PROJECT_DIR/dummy_pyproject.toml ./python_env_builddir/pyproject.toml
+      (cd python_env_builddir && poetry install --no-interaction --no-root)
```

Comment:
Why is this inside parentheses? I 'm [reading](https://ss64.com/bash/syntax-brackets.html):

> Placing a list of commands between parentheses causes a [subshell](https://unix.stackexchange.com/a/138498) to be created, and each of the commands in list to be executed in that subshell, without removing non-[exported](https://ss64.com/bash/export.html) variables.
Since the list is executed in a subshell, variable assignments do not remain in effect after the subshell completes.

But I don't see the benefit since the first command is just a `cd` command.

Reply:
We can remove them.  I didn't add this myself, but I believe the intent is so we don't need to `cd` back to the starting dir after.  For example:

```
mkdir -p /tmp/start
mkdir -p /tmp/another_location
cd /tmp/start

(cd /tmp/another_location && echo "in $(pwd)")
# in /tmp/another_location
echo "in $(pwd)"
# in /tmp

cd /tmp/another_location && echo "in $(pwd)"
# in /tmp/another_location
echo "in $(pwd)"
# in /tmp/another_location
```

We can switch it to another method if that's better though.  Could be that we `cd ..` after.  This is just a way where we don't need to care about it as much

Reply:
oh good point actually. Let's keep it as is